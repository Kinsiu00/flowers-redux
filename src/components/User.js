import React, {Component} from 'react';

class User extends Component {
    
    handleCurrentUser = () => {
        this.props.setCurrentUser(this.props.eachUser)
    }

    render() {
        const {first, last} = this.props.eachUser
        return (
            <p className='clickMe' onClick={this.handleCurrentUser}>{last}, {first}</p>
        )
    }
}

export default User;