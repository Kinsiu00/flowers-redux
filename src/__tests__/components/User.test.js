import React from 'react';
import { shallow } from 'enzyme';

import User from '../../components/User';

describe('<User>', () => {
    test('renders User', () => {
        const wrapper = shallow(<User eachUser={[]}/>)

        expect(wrapper).toMatchSnapshot()
    })
})

describe("<User />", () => {
    describe("onClick()", () => {
      test("successfully fires off click function", () => {
        const mockOnClick = jest.fn();
        const wrapper = shallow(
          <User eachUser={['test']} onClick={jest.fn()}/>
        );
        const component = wrapper
        component.find(".clickMe").simulate("click");
        expect(mockOnClick.mock.calls.length).toEqual(1);
      });
    });
  });
