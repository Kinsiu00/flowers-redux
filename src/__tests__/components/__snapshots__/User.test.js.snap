// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User> renders User 1`] = `
<p
  className="clickMe"
  onClick={[Function]}
>
  , 
</p>
`;
